src/olympia/addons/tasks.py

```diff
@@ -281,3 +281,32 @@ def delete_addons(addon_ids, with_deleted=False, **kw):
     else:
         for addon in addons:
             addon.delete(send_delete_email=False)
+
+
+@task
+@use_primary_db
+def update_addon_hotness(averages, frozen_ids):
+    log.info("[%s] Updating add-ons hotness scores.", (len(averages)))
+
+    addons = (
+        Addon.objects.filter(guid__in=averages.keys())
+        .filter(status__in=amo.REVIEWED_STATUSES)
+        .exclude(id__in=frozen_ids)
+        .no_transforms()
+    )
+
+    for addon in addons:
+        average = averages.get(addon.guid)
+        this = average['avg_this_week']
+        three = average['avg_three_weeks_before']
+
+        # Update the hotness score but only update hotness if necessary. We
+        # don't want to cause unnecessary re-indexes.
+        threshold = 250 if addon.type == amo.ADDON_STATICTHEME else 1000
+        if this > threshold and three > 1:
+            hotness = (this - three) / float(three)
+            if addon.hotness != hotness:
+                addon.update(hotness=hotness)
+        else:
+            if addon.hotness != 0:
+                addon.update(hotness=0)
```

Comment:
Not a problem with this commit specifically, but can an addon suddenly drop out of the bigquery response?  e.g. guid `foo@baa` was calculated with a hotness of 500 yesterday but today it's not in the bigquery response so it's not updated to 0?

I'm wondering if we need something that mops up any addon with a non-0 hotness that wasn't in bigquery?

Reply:
If an add-on isn't used at all anymore, yes this could probably happen. cc @diox

Reply:
Yes that's definitely an issue that needs to be addressed. The previous version of the cron handled that by querying the ids for all add-ons in the database and then only fetching update data for each chunk when looping over them.

Reply:
Should we have something like this before creating the tasks?

```
Addon.objects.filter(status__in=amo.REVIEWED_STATUSES).filter(hotness__gt=0).exclude(guid__in=averages.keys()).update(hotness=0)
```

Reply:
Probably not a good idea as it would kill hotness for everything while the tasks are being processed - and depending on the order in which they are processed, we'd see a bunch of add-ons with low hotness while the rest is being processed.